tests/Command/ListUsersCommandTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Tests\Command;
+
+use App\Command\ListUsersCommand;
+
+class ListUsersCommandTest extends AbstractCommandTest
+{
+    /**
+     * @dataProvider maxResultsProvider
+     *
+     * This test verifies the amount of data is right according to the given parameter max results.
+     */
+    public function testListUsers(int $maxResults): void
+    {
+        $tester = $this->executeCommand(
+            ['--max-results' => $maxResults]
+        );
+
+        $emptyDisplayLines = 5;
+        $this->assertSame($emptyDisplayLines + $maxResults, mb_substr_count($tester->getDisplay(), "\n"));
+    }
+
+    public function maxResultsProvider(): ?\Generator
+    {
+        yield [1];
+        yield [2];
+    }
+
+    /**
+     * @dataProvider sendEmailProvider
+     *
+     * This test ensure that the command actually sends an email when the parameter is specified
+     */
+    public function testItSendsAnEmail(bool $sendAndEmail): void
+    {
+        $this->executeCommand(
+            // these are the arguments (only 1 is passed, the rest are missing)
+            $sendAndEmail ? ['--send-to' => 'john.doe@symfony.com'] : []
+        );
+
+        $this->assertEmailCount($sendAndEmail ? 1 : 0);
+    }
+
+    public function sendEmailProvider(): ?\Generator
+    {
+        yield [true];
+        yield [false];
+    }
```

Comment:
IMHO, it's better to have 2 different tests for this, instead of using data provider.

```suggestion
    public function testItSendsNoEmailByDefault(): void
    {
        $this->executeCommand([]);

        $this->assertEmailCount(0);
    }

    public function testItSendsAnEmailIfOptionProvided(): void
    {
        $this->executeCommand(['--send-to' => 'john.doe@symfony.com']);

        $this->assertEmailCount(1);
    }
```